MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -35,6 +35,8 @@ struct EventDetails {
     var batteryPluggedIn: Bool
     var batteryLevel: Float
     var applicationState: UIApplicationState
+    var userAbsoluteDistanceToDestination: CLLocationDistance?
+    var locationEngine: CLLocationManager
```

Comment:
Let's make this a `CLLocationManager.Type` so we don't copy the whole location manager and keep an extra ref to it.